packages/library/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -106,5 +157,9 @@ class Realm private constructor(configuration: RealmConfiguration, dbPointer: Na
      */
     public override fun close() {
         super.close()
+        // TODO There is currently nothing that tears down the dispatcher
```

Comment:
Looks like tearing down the writer will require some thinking. I tried to do this for fun:

    // Verify that if multiple writes are queued up, the queue is cleared if
    // the Realm is closed in between.
    @Suppress("invisible_member")
    @Test
    fun writeClearQueueIfRealmIsClosed() = runBlocking {
        realm.write {
            val child = this.copyToRealm(Child()).apply { this.name = "Foo" }
            SystemClock.sleep(1000)
            realm.close()
            child
        }

        realm.write {
            this.copyToRealm(Child()).apply { this.name = "Bar" }
        }
        realm = Realm.open(realm.configuration)
        assertEquals(1, realm.objects<Child>().size)
    }
And it deadlocks the test.

Thinking about this a little more, then just closing the Realm isn't enough. We first need to close down all launched jobs, after which we can close the Realm. Possible by combining the dispatcher with our own supervisor, but I would need to read up on this a little more.

Reply:
I am not sure it makes sense to try to coordinate closes with the write blocks. We could try to cascade some signalling into the users job, but as cancelling is corporative anyway I don't think it would make that much sense anyway. I assume people would expect a `close` to rollback current transactions, so coordinating the actual close would basically just waste time performing operations that would be rolled back anyway. So, I therefore just close the realm when calling `close`, but eliminated the option of closing from inside a write transaction.

Reply:
I tried having an internal scope, but didn't find an easy way to cross that with the users scope so that cancelling on users scope properly cancelled that transaction. Maybe we can improve if we find the right hammer.